query-engine/prisma-models/src/builders/internal_dm_builder.rs

```diff
@@ -61,7 +65,7 @@ fn model_field_builders(model: &dml::Model, relations: &[RelationPlaceholder]) -
                 Some(FieldBuilder::Relation(RelationFieldBuilder {
                     name: rf.name.clone(),
                     arity: rf.arity,
-                    relation_name: relation.name.clone(),
+                    relation_name: schema.db.walk(relation.id).relation_name().to_string(),
```

Comment:
I'm starting to grasp the use of walkers to rehidrate data from identifiers.

Reply:
yes as long as you have an identifier and a `ParserDatabase`, you can find out absolutely anything you need to know, schema-wise.